Resources/config/mongodb.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+use Doctrine\Bundle\MongoDBBundle\Loader\SymfonyFixturesLoader;
+use Doctrine\Bundle\MongoDBBundle\ManagerConfigurator;
+use Doctrine\Bundle\MongoDBBundle\ManagerRegistry;
+use Doctrine\Bundle\MongoDBBundle\Mapping\Driver\XmlDriver;
+use Doctrine\Bundle\MongoDBBundle\Repository\ContainerRepositoryFactory;
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Doctrine\ODM\MongoDB\Mapping\Driver\AttributeDriver;
+use Doctrine\ODM\MongoDB\Tools\ResolveTargetDocumentListener;
+use Doctrine\Persistence\Mapping\Driver\MappingDriverChain;
+use ProxyManager\Proxy\GhostObjectInterface;
+use Symfony\Bridge\Doctrine\ContainerAwareEventManager;
+use Symfony\Bridge\Doctrine\Security\User\EntityUserProvider;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\abstract_arg;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $containerConfigurator->parameters()
+        ->set('doctrine_mongodb.odm.cache.array.class', 'Doctrine\Common\Cache\ArrayCache')
+        ->set('doctrine_mongodb.odm.cache.apc.class', 'Doctrine\Common\Cache\ApcCache')
+        ->set('doctrine_mongodb.odm.cache.apcu.class', 'Doctrine\Common\Cache\ApcuCache')
+        ->set('doctrine_mongodb.odm.cache.memcache.class', 'Doctrine\Common\Cache\MemcacheCache')
+        ->set('doctrine_mongodb.odm.cache.memcache_host', 'localhost')
+        ->set('doctrine_mongodb.odm.cache.memcache_port', 11211)
+        ->set('doctrine_mongodb.odm.cache.memcache_instance.class', 'Memcache')
+        ->set('doctrine_mongodb.odm.cache.xcache.class', 'Doctrine\Common\Cache\XcacheCache')
+        ->set('doctrine_mongodb.odm.metadata.driver_chain.class', MappingDriverChain::class)
+        ->set('doctrine_mongodb.odm.metadata.attribute.class', AttributeDriver::class)
+        ->set('doctrine_mongodb.odm.metadata.xml.class', XmlDriver::class);
```

Comment:
Related to #811, the parameters should be deprecated here.

Reply:
This is partially related to https://github.com/doctrine/DoctrineMongoDBBundle/pull/804, these parameters are apparently used in the AbstractDoctrineExtension, this is the code related about cache:

https://github.com/symfony/symfony/blob/e0df5c3d9c0281cdfe3885281145e2065b859d32/src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php#L340-L406

And the related to the metadata:

https://github.com/symfony/symfony/blob/e0df5c3d9c0281cdfe3885281145e2065b859d32/src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php#L193-L228

Reply:
Even if we stop using them, we have to keep and deprecate them in 2.x, as they would be used in applications.

Reply:
I was thinking that since these parameters seem to be used for internal configuration, maybe Symfony can try to fetch them first as [build parameter](https://github.com/symfony/symfony/pull/47680) and fallback to "normal", I'll see if I have time tomorrow to open a PR there for that.